arch/arm/src/nrf52/nrf52_i2c.c

```diff
@@ -373,7 +374,9 @@ static int nrf52_i2c_transfer(struct i2c_master_s *dev,
           /* Write number of bytes in TXD buffer */
 
           regval = priv->dcnt;
-          nrf52_i2c_putreg(priv, NRF52_TWIM_TXMAXCNT_OFFSET, regval);
+          nrf52_i2c_putreg(priv, NRF52_TWIM_TXDMAXCNT_OFFSET, regval);
+
+          nxsig_usleep(1);
```

Comment:
I can't find a specific erratum, and unfortunately I can't find anything in the NRF52 registers to check instead of sleeping. I suspect it's specific to the NRF52840 and not the board, and that other boards were maybe not well tested, but I can't check that thesis.

Reply:
But what is the problem here exactly ? could you explain what didn't work ?

I have this code running on nrf52840 with i2c eeprom on >1.5k devices and it works fine, so I'm curious what problems you've noticed.

Reply:
just tested nrf52840-dk with lsm9ds1 over i2c and I don't see any issues. 

400kHz with this change makes the i2c bus inefficient:
<img width="1389" alt="image" src="https://github.com/apache/nuttx/assets/6563055/5fcd4996-8160-4a28-8cc5-19af36734f0e">

here without sleep:
<img width="1392" alt="image" src="https://github.com/apache/nuttx/assets/6563055/ae735ba2-572a-41f4-897e-708120a747e6">

<img width="1394" alt="image" src="https://github.com/apache/nuttx/assets/6563055/e9d34f1e-847f-4379-861d-4c4dacda23d8">

So I guess the problem is somewhere else and this change should not be applied to nrf52 i2c driver.

Reply:
The LASTTX event was never firing. If it works elsewhere then I see three options:

1) It's board specific, in which case we would put a board specific #ifdef around it.
2) It's something specific to the LMS9DS1, which is the only device I've tested. I can validate that once I bring up the on-board barometer, which was next on my list.
3) I've done something wrong in the board configuration. I'm struggling to figure out what, though, since it works in general with this delay.

What would you like me to do?

Reply:
For now, I'd prefer to abandon this change until we find out what the problem is. 
How did you have the `CONFIG_I2C_POLLED` option set? My tests where with `CONFIG_I2C_POLLED=n`, I forgot to check the second option.

Reply:
I tried both polled and with interrupts with the same results. I will do as you suggest and add an nrf52 specific option for now using up_udelay(). One possibility is that I've done something wrong with the GPIO pins but I doubt it since I'm matching the other boards. (You may have already noted that the I2C pin config in the NRF52840 manual does NOT work at all, unfortunately.)

Reply:
GPIOs in NRF5x are rather simple so it's hard to go wrong here.
Maybe the bootloader is setting some registers wrong and this is not being reset by NuttX initialization. But I have no idea what it could be.

Reply:
I ordered this board, so in a few days I'll try to help with this I2C problem

Reply:
@brian-doherty I managed to communicate with APDS-9660 sensor on my board (I have Nano 33 BLE Sense rev 2, so no LMS9DS1 here). 

What is missing:
1. set `VDD_ENV` high (P0.22)
2. wait some time to be sure that all sensors power up correctly (I just use `up_mdelay(20)`)

Reply:
Nice! I will make the changes this weekend.

Reply:
When I make these, do you want me to squash again, or should I leave them separate so you can examine them separately?

Reply:
Please squash all commits related to the board support and keep fixes for arch/nrf52 as separate commits. Then the git history will be prettier and it will be easier to create release notes.